terraform/node/ec2/asg/main.tf

```diff
@@ -0,0 +1,281 @@
+  ssh_key_name        = aws_key_pair.aws_ssh_key.key_name
+  private_key_content = tls_private_key.ssh_key.private_key_pem
+}
+
+data "aws_ami" "ami" {
+  owners = ["amazon"]
+  most_recent      = true
+  filter {
+    name   = "name"
+    values = ["al20*-ami-minimal-*-x86_64"]
+  }
+  filter {
+    name   = "state"
+    values = ["available"]
+  }
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+  filter {
+    name   = "image-type"
+    values = ["machine"]
+  }
+
+  filter {
+    name   = "root-device-name"
+    values = ["/dev/xvda"]
+  }
+
+  filter {
+    name   = "root-device-type"
+    values = ["ebs"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+}
+
+resource "aws_launch_configuration" "launch_configuration" {
+  image_id      = data.aws_ami.ami.id
+  instance_type = "t3.micro"
+  key_name = local.ssh_key_name
+  associate_public_ip_address = true
+  iam_instance_profile = "APP_SIGNALS_EC2_TEST_ROLE"
+  security_groups = [aws_default_vpc.default.default_security_group_id]
+
+  user_data = <<-EOF
+    #!/bin/bash
+
+    # Set up environment
+    sudo yum install nodejs unzip wget tmux aws-cli -y
+
+    echo "Node version in use: $(node -v)"
+
+    # Copy in CW Agent configuration
+    agent_config='${replace(replace(file("./amazon-cloudwatch-agent.json"), "/\\s+/", ""), "$REGION", var.aws_region)}'
+    echo $agent_config > amazon-cloudwatch-agent.json
+
+    # Get and run CW agent rpm
+    ${var.get_cw_agent_rpm_command}
+    sudo rpm -U ./cw-agent.rpm
+    sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -s -c file:./amazon-cloudwatch-agent.json
+
+    # Get and run the sample application with configuration
+    aws s3 cp ${var.sample_app_zip} ./node-sample-app.zip
+    unzip -o node-sample-app.zip
+
+    # Enter appropriate service folder
+    cd frontend-service
+
+    # Install sample application
+    npm install
+
+    # Get ADOT instrumentation and install it
+    ${var.get_adot_instrumentation_command}
+
+    # Set up application tmux screen so it keeps running after closing the SSH connection
+    tmux new-session -d -s frontend
+
+    # Export environment variables for instrumentation
+    tmux send-keys -t frontend 'export OTEL_METRICS_EXPORTER=none' C-m
+    tmux send-keys -t frontend 'export OTEL_TRACES_EXPORTER=otlp' C-m
+    tmux send-keys -t frontend 'export OTEL_AWS_APPLICATION_SIGNALS_ENABLED=true' C-m
+    tmux send-keys -t frontend 'export OTEL_AWS_APPLICATION_SIGNALS_EXPORTER_ENDPOINT=http://localhost:4316/v1/metrics' C-m
+    tmux send-keys -t frontend 'export OTEL_EXPORTER_OTLP_TRACES_ENDPOINT=http://localhost:4316/v1/traces' C-m
+    tmux send-keys -t frontend 'export OTEL_EXPORTER_OTLP_TRACES_PROTOCOL=http/protobuf' C-m
+    tmux send-keys -t frontend 'export OTEL_EXPORTER_OTLP_METRICS_PROTOCOL=http/protobuf' C-m
+    tmux send-keys -t frontend 'export OTEL_NODE_DISABLED_INSTRUMENTATIONS=fs,dns,express' C-m
```

Comment:
Can we leave comment here (and the other 3 places this is set) to explain why we add `express` to disabled instrumentation list to reduce the overhead in the testing? I recall there is some random order introduced by express that affects the validator.

Reply:
Added a comment above each of the tmux sections where this is used.